src/VisualStudio/Core/Def/Implementation/ProjectSystem/CPS/IProjectShim.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.CodeAnalysis;
+using Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.ProjectSystem
+{
+    internal interface IProjectShim : IAnalyzerHost
```

Comment:
> Why does it derive from IAnalyzerHost?

Basically all the APIs on IAnalyzerHost are needed to be invoked by CPS as they deal with items (ruleset, additional files, and analyzer references). Alternatively, we can avoid deriving the interface from IAnalyzerHost and have CPS explicitly MEF import IAnalyzerHost to forward analyzer host related item changes.